src/libraries/Microsoft.PowerFx.Core/Public/Values/RecordValue.cs

```diff
@@ -198,9 +198,9 @@ internal async Task<FormulaValue> GetFieldAsync(FormulaType fieldType, string fi
         /// <param name="fieldName">Name of the field.</param>        
         /// <param name="result"></param>
         /// <returns>true if field is present, else false.</returns>
-        protected abstract bool TryGetField(FormulaType fieldType, string fieldName, out FormulaValue result);
-        
-        protected virtual Task<(bool Result, FormulaValue Value)> TryGetFieldAsync(FormulaType fieldType, string fieldName, CancellationToken cancellationToken)
+        protected abstract bool TryGetField(FormulaType fieldType, string fieldName, out FormulaValue result);
+
+        protected internal virtual Task<(bool Result, FormulaValue Value)> TryGetFieldAsync(FormulaType fieldType, string fieldName, CancellationToken cancellationToken)
```

Comment:
>internal [](http://example.com/codeflow?start=18&length=8)

do we need this? Can't they just call the public method (GetFieldAsync)?

Reply:
No, this generates an error (like for TryGetField)

Reply:
Will use GetField instead